### PR TITLE
Refactoring bytecode_module to use flatcc instead of flatbuffers C++.

### PR DESCRIPTION
**Pull request by benvanik**
This improves our validation, removes manually-synchronized redundancy
with the schema for function descriptors, and optimizes a few things like
redundant type resolution.

Required a fix to cc_embed_data; the way it stores files as strings does not
guarantee alignment. The Flatbuffers C++ API papers over this and allows
for unaligned accesses, but flatcc doesn't (which is good); now all files
will be pointer-aligned.

Progress on #898.

### CODE REVIEW

**Review by ScottTodd — COMMENTED** [review]
Code LGTM, once we've figured out how to add the flatcc dep. I'd prefer to sequence this change after the dependency is added and fully merged though.

File: iree/vm/bytecode_module_impl.h
Comment (antiagainst) [inline iree/vm/bytecode_module_impl.h:21]: FYI: This will also have problem with Android, where we don't have `cstdalign`. `stdalign.h` is fine, at least with ndk-r21d.

**Review by ScottTodd — APPROVED** [review]

**Comment by ScottTodd** [comment]
> Code LGTM, once we've figured out how to add the flatcc dep. I'd prefer to sequence this change after the dependency is added and fully merged though.

Should be ready now.

File: iree/vm/bytecode_module_impl.h
Comment (antiagainst) [reply iree/vm/bytecode_module_impl.h:21]: https://github.com/google/iree/commit/5589fd2c6f83fbea62dd885c57fad1b04ec40948 is a possible fix to make it compile. Ben might want to see whether it makes sense though.

**Comment by GMNGeoffrey** [comment]
Can we go ahead and merge this? I'm waiting for some old PRs to get merged before I do more fiddling with the CI directory structure

**Comment by antiagainst** [comment]
This now breaks Android cross compilation.. Ben, have you checked 5589fd2 and see whether it make sense to have it as a patch?

**Pull request merged**
